utils/slack_api.py

```diff
@@ -14,13 +14,14 @@ class SlackApi(object):
     """Wrapper around Slack API calls"""
 
     def __init__(self, token, settings=None, init_usergroups=True,
-                 **chat_kwargs):
+                 workspace_name=None, **chat_kwargs):
         slack_token = secret_reader.read(token, settings=settings)
         self.sc = SlackClient(slack_token)
         self.results = {}
         self.chat_kwargs = chat_kwargs
         if init_usergroups:
             self.initiate_usergroups()
+        self.workspace_name = workspace_name or ''
```

Comment:
Maybe we should fail if workspace_name is not passed instead of possibly passing an empty string down the line in slack_map or when we build states?

Reply:
I agree, it looks weird that it can be empty...

Reply:
make workspace_name a required argument for SlackApi init